tests/Integration/ScriptsTest.php

```diff
@@ -186,6 +186,80 @@ public function test_enqueue_js_tracker(): void {
 		self::assertEquals( 1, count( $wp_scripts->registered['wp-parsely-loader']->extra ) );
 	}
 
+	/**
+	 * Verifies that tracker scripts are not loading for drafted posts
+	 *
+	 * @covers \Parsely\Scripts::enqueue_js_tracker
+	 *
+	 * @uses Parsely\Parsely::api_key_is_set
+	 * @uses Parsely\Parsely::get_api_key
+	 * @uses Parsely\Parsely::get_options
+	 * @uses Parsely\Parsely::get_tracker_url
+	 * @uses Parsely\Scripts::__construct
+	 * @uses Parsely\Scripts::register_scripts
+	 *
+	 * @group scripts
+	 */
+	public function test_should_not_enqueue_tracker_scripts_for_drafted_posts(): void {
+		$this->set_admin_user();
+		$this->go_to_new_post( 'draft' );
+		
+		self::$scripts->register_scripts();
+		self::$scripts->enqueue_js_tracker();
+
+		// Verify that tracker script is registered and enqueued.
+		$this->assert_script_statuses(
+			'wp-parsely-tracker',
+			array( 'registered' ),
+			array( 'enqueued' )
```

Comment:
If the test method is to check the tracking script is not enqueued, why does this line suggest it's looking for an enqueued state?

Reply:
It is not looking for enqueued state instead it is looking for assert false on `array( 'enqueued' )`. Please have look at [assert_script_statuses](https://github.com/Parsely/wp-parsely/blob/changes/avoid-tracking-views-of-published-post-in-preview-mode/tests/Integration/ScriptsTest.php#L566) function signature.

Reply:
Oh, of course 🤦🏻‍♂️

Reply:
In which case the comments above these calls in the various test methods need reviewing and fixing.

i.e.
```diff
- // Verify that tracker script is registered and enqueued.
+ // Verify that tracker script is registered but not enqueued.
```

Reply:
I think we shouldn't have those comments because those are just explaining the already good written code. So removing these.

Reply:
I had the same confusion as Gary. Boolean arguments (or arguments that have Boolean behavior, in this case) are inherently confusing. No need to tackle in this PR, but we should try to avoid them in the future.

Reply:
@chriszarate here I am using the already available `assert_script_statuses` function ...... so in future how can we avoid it if we need to use the same function?

Should we create new functions with more clear names?

Reply:
We can create another function if we find this one confusing, but let's leave this for another PR. When we create the new function, we can also replace the old calls with the new ones.

Reply:
I agree with Chris and Alex (and the removal of the comments). Another PR is the better place to address this.

Move the boolean aspect to the assertion name:
```
$this->assert_script_is_registered( 'wp-parsely-tracker' );
$this->assert_script_is_not_registered( 'wp-parsely-tracker' );
$this->assert_script_is_enqueued( 'wp-parsely-tracker' );
$this->assert_script_is_not_enqueued( 'wp-parsely-tracker' );
```